# Project 5 -- Jonathan Contreras

**TA Help:**
    
**Collaboration:** 

## Question 1

In [1]:
%%bash
# part a) extracting the data in the 9th column (UniqueCarrier), sorting it, counting unique data, and sorting it numerically
cat /anvil/projects/tdm/data/flights/subset/[12]*.csv | cut -d, -f9 | sort | uniq -c | sort -n

     22 UniqueCarrier
  70622 ML (1)
  83617 PS
 154381 AQ
 208420 TZ
 230824 HA
 271876 F9
 316167 PA (1)
 347210 9E
 682728 B6
 684812 YV
 693047 DH
 873957 PI
 919785 EA
1090310 FL
1338503 OH
1508980 EV
2114556 XE
2718878 OO
2776804 AS
3632062 MQ
3636682 HP
3757747 TW
7950035 CO
10067774 NW
13004338 UA
13775480 US
14584281 AA
15173234 WN
16247348 DL


In [2]:
%%bash
# part b) extracting only the number of distinct data in the 11th column. wc -l gets the line count of the list.
cat /anvil/projects/tdm/data/flights/subset/[12]*.csv | cut -d, -f11 | sort |uniq | wc -l

12734


Markdown notes and sentences and analysis written here.

## Question 2

In [3]:
%%bash
# extracting the head data of the csv
cat /anvil/projects/tdm/data/8451/The_Complete_Journey_2_Master/5000_transactions.csv | head

BASKET_NUM              ,HSHD_NUM        ,PURCHASE_,PRODUCT_NUM                 ,     SPEND,     UNITS,STORE_R,  WEEK_NUM,YEAR
000024                  ,1809            ,03-JAN-16,5817389                     ,      -1.5,        -1,SOUTH  ,         1,2016
000024                  ,1809            ,03-JAN-16,5829886                     ,      -1.5,        -1,SOUTH  ,         1,2016
000034                  ,1253            ,03-JAN-16,0539501                     ,      2.19,         1,EAST   ,         1,2016
000060                  ,1595            ,03-JAN-16,5260099                     ,       .99,         1,WEST   ,         1,2016
000060                  ,1595            ,03-JAN-16,4535660                     ,       2.5,         2,WEST   ,         1,2016
000168                  ,3393            ,03-JAN-16,5602916                     ,       4.5,         1,SOUTH  ,         1,2016
000199                  ,4558            ,03-JAN-16,5683532                     ,      3.49,         1,SOUTH  ,

In [4]:
%%bash
# extracting only the number of distinct PRODUCT_NUM values
cat /anvil/projects/tdm/data/8451/The_Complete_Journey_2_Master/5000_transactions.csv | cut -d, -f4 | sort | uniq | wc -l

151142


Markdown notes and sentences and analysis written here.

## Question 3

In [5]:
%%bash 
# extracting the head of the data
cat /anvil/projects/tdm/data/election/itcont*.txt | head

C00078279|A|M11|P|80031492155|22Y||MCKENNON, K R|MIDLAND|MI|00000|||10031979|400|||||CONTRIBUTION REF TO INDIVIDUAL|3062020110011466469
C00078279|A|M11||79031415137|15||OREFFICE, P|MIDLAND|MI|00000|DOW CHEMICAL CO||10261979|1500||||||3061920110000382948
C00078279|A|M11||79031415137|15||DOWNEY, J|MIDLAND|MI|00000|DOW CHEMICAL CO||10261979|300||||||3061920110000382949
C00078279|A|M11||79031415137|15||BLAIR, E|MIDLAND|MI|00000|DOW CHEMICAL CO||10261979|1000||||||3061920110000382950
C00078287|A|Q1||79031231889|15||BLANCHARD, JOHN A|CHICAGO|IL|60685|||03201979|200||||||3061920110000383914
C00078287|A|Q1||79031231889|15||CRAMER, JOHN H|CHICAGO|IL|60685|||02281979|200||||||3061920110000383915
C00078287|A|Q1||79031231889|15||MCHUGH, KEVIN|CHICAGO|IL|60685|||03051979|200||||||3061920110000383916
C00078287|A|Q1||79031231889|15||NOHA, EDWARD J|CHICAGO|IL|60685|||03121979|300||||||3061920110000383917
C00078287|A|Q1||79031231889|15||RYCROFT, DONALD C|CHICAGO|IL|60685|||03191979|200||||||30619201100

In [6]:
%%bash
# part a) extracting the number of distinct IDs altogether
cat /anvil/projects/tdm/data/election/itcont*.txt | cut -d, -f1 | sort | uniq | wc -l

186421390


In [7]:
%%bash
# part b) extracting the cID with the largest donations and how many donations they made.
cat /anvil/projects/tdm/data/election/itcont*.txt | cut -d '|' -f1 | sort | uniq -c | sort -nr | head -n 1

50696691 C00401224


The committee ID that received the largest number of donations was C00401224 with 50696691 received donations.

## Question 4

In [12]:
%%bash
# trial and errors: 
# grep -E "Lafayette|West Lafayette" /anvil/projects/tdm/data/beer/breweries.csv
# grep -E 'Lafayette|West Lafayette' /anvil/projects/tdm/data/beer/breweries.csv
# grep -E 'Lafayette|West Lafayette' /anvil/projects/tdm/data/beer/breweries.csv | grep -i 'Indiana'
# grep -E "Lafayette | West Lafayette" | grep "IN" /anvil/projects/tdm/data/beer/breweries.csv | wc -l
# cat /anvil/projects/tdm/data/beer/breweries.csv | grep -E "Lafayette|West Lafayette" | grep ",IN" | wc -l 

# -E allows operations like '|', which represents 'or', to be used
cat /anvil/projects/tdm/data/beer/breweries.csv | grep -E "Lafayette|West Lafayette" | grep ",IN"

51616,Brokerage Brewing Company,West Lafayette,IN,US,No notes at this time.,"Brewery, Bar, Beer-to-go"
30699,Village Bottle Shoppe #3,West Lafayette,IN,US,No notes at this time.,Store
40401,Harry's Chocolate Shop,West Lafayette,IN,US,No notes at this time.,"Bar, Eatery"
16648,Knickerbocker Saloon,Lafayette,IN,US,No notes at this time.,"Bar, Eatery"
52229,Teays River Brewing & Public House,Lafayette,IN,US,No notes at this time.,"Brewery, Bar, Eatery"
32945,The Pint,West Lafayette,IN,US,No notes at this time.,Bar
21952,People's Brewing Company,Lafayette,IN,US,No notes at this time.,"Brewery, Bar, Beer-to-go"
45760,Walt's Other Pub,Lafayette,IN,US,No notes at this time.,"Bar, Eatery"
14471,Scotty's Brewhouse,West Lafayette,IN,US,No notes at this time.,"Bar, Eatery, Beer-to-go"
22332,Nine Irish Brothers,West Lafayette,IN,US,No notes at this time.,"Bar, Eatery"
14469,Bar Barry Liquors - West,West Lafayette,IN,US,No notes at this time.,Store
31172,Jake's Roadhouse & Garage,West Lafayette,IN,

Markdown notes and sentences and analysis written here.

## Question 5

In [13]:
%%bash
# extracting the sorted list of 25 words that occurred the most in the file and the number of times that word occurred
cat /anvil/projects/tdm/data/icecream/combined/reviews.csv | tr ' ' '\n' | sort | uniq -c | sort -n | tail -25

   5470 like
   5494 chocolate
   5777 not
   5845 love
   5928 have
   6274 with
   6409 that
   6751 so
   7375 flavor
   7495 but
   7551 for
   9981 was
  10329 in
  10598 my
  12940 cream
  15065 it
  15300 this
  17072 of
  17778 to
  18481 is
  18688 ice
  19480 a
  26848 and
  29090 I
  34937 the


Markdown notes and sentences and analysis written here.

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.